SA3/qa_web/views.py

```diff
@@ -29,6 +29,41 @@ def index(request):
     return render(request, 'qa_web/index.html', context={'questions': Questions.objects.all()})
 
 
+@login_required(login_url='/login/')
+def edit_profile(request):
+    if request.method=='GET':
+        form = UserProfile()
+        # return render_to_response('qa_web/UserProfile.html', RequestContext(request, {'form': form, }))
+        return render(request,'qa_web/EditUserProfile.html', context={'form': form})
+    else:
+        form = UserProfile(request.POST)
+        # validating the form occurs below and then saving the results entered by the user
+        if form.is_valid():
+            user = request.user
+
+            user.first_name = request.POST['prename']
+            user.last_name = request.POST['surname']
+            user.age = request.POST['age']
+            user.birthday = request.POST['birthday']
```

Comment:
Instead of assigning a string from `request.post` to this `User` field, you can read up on using the Django form's value, which will automatically recognize any Date objects.

Because right now, this will throw an exception on server side if you do not have the specific YYYY-MM-DD format.

Reply:
The specific date format is now depicted to the user, as an error message so that he/she does not make that mistake.